### PR TITLE
[SSL Simulation Protocol] Command line args and config

### PR DESCRIPTION
**Pull request by jsn0716**
### Description

Adds dynamic parameters for SSL simulation command line args and config for [SSL simulation protocol support PR](https://github.com/UBC-Thunderbots/SSL-Simulation-Protocol-Support/pull/11)

### Testing Done

Local testing

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/shared/parameter/config_definitions/ssl_simulation_config.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+- int:
+    name: team_channel
```
Comment (akhilveeraghanta) [inline src/shared/parameter/config_definitions/ssl_simulation_config.yaml:2]: curious why there are two yamls? the second seems to contain all the parameters in this file?

**Review by jonathanlew — COMMENTED** [review]
nothing to add beyond what's already been said

File: src/shared/parameter/config_definitions/ssl_simulation_config.yaml (hunk 1 above)
Comment (jsn0716) [reply src/shared/parameter/config_definitions/ssl_simulation_config.yaml:2]: [`ssl_simulation_main.cpp`](https://github.com/UBC-Thunderbots/SSL-Simulation-Protocol-Support/pull/11/files) shows how the configs are used, one is for command line args and the other is the config actually given to `SslSimulationProtocolManager`

File: src/shared/parameter/config_definitions/ssl_simulation_main_command_line_args.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        - "div_a"
+        - "div_b"
+    description: >-
+        Which ssl_division to configure the simulator to. Changes the field size and the number
+        of initial robots.
+
+- enum:
+    name: team_colour
+    enum: TeamColour
+    value: YELLOW
```
Comment (kelvinkoon) [inline src/shared/parameter/config_definitions/ssl_simulation_main_command_line_args.yaml:27]: Nit: Does it matter if we have an options field like we did for `ssl_division` (except for `BLUE` and `YELLOW`)

**Review by kelvinkoon — COMMENTED** [review]
Nicely done, nothing to add besides the existing comments.

File: src/shared/parameter/config_definitions/ssl_simulation_main_command_line_args.yaml (hunk 2 above)
Comment (jsn0716) [reply src/shared/parameter/config_definitions/ssl_simulation_main_command_line_args.yaml:27]: Shouldn't really make a difference - the enum only allows for yellow and blue anyways

**Review by jonathanlew — DISMISSED** [review]
:shipit:

**Comment by alinas2000** [comment]
LGTM! Really nit: some descriptions in the yaml files end with a "." , some don't.

File: src/shared/parameter/config_definitions/ssl_simulation_config.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    min: 0
+    max: 15
+    value: 0
+    description: >-
+      The multicast channel to use to communicate with the blue robots
```
Comment (amrm3lm) [inline src/shared/parameter/config_definitions/ssl_simulation_config.yaml]: is this meant to be for friendly robots, regardless whether they are blue or yellow?

**Review by amrm3lm — COMMENTED** [review]
lgtm, just one comment

**Review by akhilveeraghanta — DISMISSED** [review]

File: src/shared/parameter/config_definitions/ssl_simulation_config.yaml (hunk 3 above)
Comment (jsn0716) [reply src/shared/parameter/config_definitions/ssl_simulation_config.yaml]: Oops supposed to be friendly robots, fixed

**Review by jonathanlew — APPROVED** [review]

**Review by akhilveeraghanta — APPROVED** [review]

**Pull request merged**